tests/Bridge/OpenAI/DallE/ModelClientTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Tests\Bridge\OpenAI\DallE;
+
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE;
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE\Base64Image;
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE\GeneratedImagesResponse;
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE\ModelClient;
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE\UrlImage;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Small;
+use PHPUnit\Framework\Attributes\Test;
+use PHPUnit\Framework\Attributes\UsesClass;
+use PHPUnit\Framework\TestCase;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface as HttpResponse;
+
+#[CoversClass(ModelClient::class)]
+#[UsesClass(DallE::class)]
+#[UsesClass(HttpResponse::class)]
+#[UsesClass(HttpClientInterface::class)]
+#[UsesClass(UrlImage::class)]
+#[UsesClass(Base64Image::class)]
+#[UsesClass(GeneratedImagesResponse::class)]
+#[Small]
+class ModelClientTest extends TestCase
+{
+    #[Test]
+    public function itIsSupportingTheCorrectModel(): void
+    {
+        $modelClient = new ModelClient(self::createStub(HttpClientInterface::class), 'sk-api-key');
+
+        self::assertTrue($modelClient->supports(new DallE(), 'foo'));
+    }
+
+    #[Test]
+    public function itIsExecutingTheCorrectRequest(): void
+    {
+        $httpClient = $this->createMock(HttpClientInterface::class);
+        $httpClient->expects(self::once())
+            ->method('request')
+            ->with('POST', 'https://api.openai.com/v1/images/generations', [
+                'auth_bearer' => 'sk-api-key',
+                'json' => [
+                    'model' => DallE::DALL_E_2,
+                    'prompt' => 'foo',
+                    'n' => 1,
+                    'response_format' => 'url',
+                ],
+            ]);
```

Comment:
Lets use `MockHttpClient` here directly

Reply:
I still see no advantage in this change. Maybe you are able to describe it to me? Still the only advantage that comes in my mind from the documentation is this:

> However, using [MockResponse](https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/HttpClient/Response/MockResponse.php) allows simulating chunked responses and timeouts:
See: https://symfony.com/doc/current/http_client.html#http-client-and-responses 

And that is not what i am doing here. I will change it this time for this single test case but in general it would be nice to understand why you are having the opinion it would be better utilized then what i wrote with the mock. The assertions just look a bit different.

Reply:
same, too many bad experience and refactoring pain with mocks - even if i feel like the MockHttpClient is that smooth.

thanks for adopting! :+1: